tests/e2e/startup/startup_test.go

```diff
@@ -157,10 +136,87 @@ var _ = Describe("Various Startup Configurations", Ordered, func() {
 		})
 
 		It("Kills the cluster", func() {
-			err := KillK3sCluster(tc.AllNodes())
+			err := e2e.KillK3sCluster(tc.AllNodes())
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+
+	Context("Verify SQLite to Etcd migration", func() {
+		It("Starts up with SQLite and checks status", func() {
+			err := StartK3sCluster(tc.AllNodes(), "", "")
+			Expect(err).NotTo(HaveOccurred(), e2e.GetVagrantLog(err))
+
+			By("CLUSTER CONFIG")
+			By("OS:" + *nodeOS)
+			By(tc.Status())
+			tc.KubeconfigFile, err = e2e.GenKubeconfigFile(tc.Servers[0].String())
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Fetching node status")
+			Eventually(func() error {
+				return tests.NodesReady(tc.KubeconfigFile, e2e.VagrantSlice(tc.AllNodes()))
+			}, "600s", "5s").Should(Succeed())
+			Eventually(func() error {
+				return tests.AllPodsUp(tc.KubeconfigFile)
+			}, "600s", "5s").Should(Succeed())
+			Eventually(func() error {
+				return tests.CheckDefaultDeployments(tc.KubeconfigFile)
+			}, "480s", "10s").Should(Succeed())
+			e2e.DumpPods(tc.KubeconfigFile)
+		})
+
+		It("Creates test resources before migration", func() {
+			createCmd := "kubectl create configmap migration-test --from-literal=test=before-migration"
+			_, err := tc.Servers[0].RunCmdOnNode(createCmd)
+			Expect(err).NotTo(HaveOccurred())
+
+			getCmd := "kubectl get configmap migration-test -o jsonpath='{.data.test}'"
+			result, err := tc.Servers[0].RunCmdOnNode(getCmd)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(result).To(ContainSubstring("before-migration"))
+		})
+
+		It("Migrates from SQLite to etcd", func() {
+			configCmd := "echo 'cluster-init: true' >> /etc/rancher/k3s/config.yaml"
+			_, err := tc.Servers[0].RunCmdOnNode(configCmd)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(e2e.RestartCluster(tc.Servers)).To(Succeed())
+			Expect(e2e.RestartCluster(tc.Agents)).To(Succeed())
+
+			Eventually(func() (string, error) {
+				cmd := "kubectl get node server-0 --show-labels"
+				return tc.Servers[0].RunCmdOnNode(cmd)
+			}, "120s", "5s").Should(ContainSubstring("node-role.kubernetes.io/etcd=true"))
```

Comment:
nit: its a label, use it to filter the node list
```suggestion
				cmd := "kubectl get node server-0 -l node-role.kubernetes.io/etcd=true"
				return tc.Servers[0].RunCmdOnNode(cmd)
			}, "120s", "5s").Should(ContainSubstring(tc.Servers[0]))
```